src/pyairios/__init__.py

```diff
@@ -77,21 +79,23 @@ async def fetch(self) -> AiriosData:
         """Get the data from all nodes at once."""
         data: dict[int, AiriosNodeData] = {}
 
-        brdg_data = await self.bridge.fetch_bridge()
-        if brdg_data["rf_address"] is None:
+        brdg_data = await self.bridge.fetch_bridge_data()
+        if brdg_data is None or brdg_data["rf_address"] is None:
             raise AiriosException("Failed to fetch node RF address")
         bridge_rf_address = brdg_data["rf_address"].value
         data[self.bridge.slave_id] = brdg_data
 
-        for node in await self.bridge.nodes():
-            if node.product_id == ProductId.VMD_02RPS78:
-                vmd = VMD02RPS78(node.slave_id, self.bridge.client)
-                vmd_data = await vmd.fetch_vmd_data()
-                data[node.slave_id] = vmd_data
-            if node.product_id == ProductId.VMN_05LM02:
-                vmn = VMN05LM02(node.slave_id, self.bridge.client)
-                vmn_data = await vmn.fetch_vmn_data()
-                data[node.slave_id] = vmn_data
+        prids = brdg_data["product_ids"]
+        if prids is not None and brdg_data["models"] is not None:
+            for _node in await self.bridge.nodes():
+                for key, _id in prids.items():
+                    if _id == _node.product_id and brdg_data["models"][key] is not None:
+                        LOGGER.debug("fetch_node_data for key: %s", key)
+                        node_module = brdg_data["models"][key].Node(
+                            _node.slave_id, self.bridge.client
+                        )
+                        node_data = await node_module.fetch_node_data()
+                        data[_node.slave_id] = node_data
```

Comment:
would be nice to print a warning if an unsupported device is bound and returned in "self.bridge.nodes()"